meta-freescale-3rdparty-mbl/conf/machine/imx6ul-pico-mbl.conf

```diff
@@ -0,0 +1,123 @@
+##############################################################################
+#SPL_BINARY=""
+UBOOT_EXTLINUX = "0"
+#UBOOT_CONFIG = ""
+#UBOOT_MACHINE = "pico-pi-imx6ul_defconfig"
+#UBOOT_SUFFIX = "bin"
+#UBOOT_BINARY = "u-boot.${UBOOT_SUFFIX}"
+#UBOOT_ENTRYPOINT = "0x87800000"
+#UBOOT_DTB_LOADADDRESS = "0x83000000"
+#UBOOT_IMAGE = "mbl-u-boot.bin"
+#UBOOT_SIGN_ENABLE = "1"
+
+# WKS_FILES is a priority ordered list of .wks files to look for (see
+# image_types_wic.bbclass). The default value of WKS_FILES has ${WKS_FILE}
+# listed first so that the user can set the .wks file by setting WKS_FILE.
+#
+# meta-freescale-3rdparty's imx6l-pico.conf sets WKS_FILES to
+# "imx-uboot-spl.wks", but that prevents specifying the .wks file in WKS_FILE.
+# Fix that here re-setting WKS_FILES to list ${WKS_FILE} first. There's not
+# much point in us giving imx-uboot-spl.wks as a second item in the list (to
+# use as a default .wks file for when WKS_FILE isn't set) because our BSP
+# doesn't use SPL.
+WKS_FILES = "${WKS_FILE}"
+
+# Remove all recipes except kernel modules recipe in fsl-bsp-release
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/classes/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-browser/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-connectivity/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-core/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-devtools/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-graphics/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-multimedia/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-security/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-utils/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-kernel/linux/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-kernel/linux-firmware/"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-kernel/kernel-modules/kernel-module-imx-gpu-viv*.bb"
+
+# Because of license restrictions we want to use fsl-eula-unpack.bbclass from
+# meta-fsl-bsp-release instead the one provided by meta-freescale.
+# Also mask firmware-imx in meta-fsl-bsp-release-mbl to avoid missing
+# target file error for bbappend file.
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/recipes-bsp/(?!firmware-qca)"
+BBMASK += "/meta-fsl-bsp-release/imx/meta-bsp/classes/(?!fsl-eula-unpack\.bbclass)"
+BBMASK += "/meta-freescale/classes/fsl-eula-unpack.bbclass"
+BBMASK += "/meta-mbl/meta-fsl-bsp-release-mbl/imx/meta-bsp/recipes-bsp/firmware-imx/"
+
+# We need to set the EULA file from meta-fsl-bsp-release instead of
+# meta-freescale/EULA
+FSL_EULA_FILE = "${OEROOT}/layers/meta-fsl-bsp-release/imx/EULA.txt"
+
+# Specify the ATF recipe to be used for this machine.
+#PREFERRED_PROVIDER_virtual/atf = "atf-${MACHINE}"
+
+# Don't use any of the firmware-imx-* packages - they're covered by a EULA that
+# prevents us doing binary releases
+MACHINE_EXTRA_RRECOMMENDS_remove = "firmware-imx-brcm"
```

Comment:
I think `firmware-imx-brcm` is not installed for this platform.
Can you please confirm if  it is the same case for `firmware-imx-vpu-imx6d`?

Reply:
I just confirmed that the `firmware-imx-brcm` is not added by default but the `firmware-imx-vpu-imx6d` is. So, we just need to remove the installation of `firmware-imx-vpu-imx6d`.

Reply:
What is strange is that the imx6ul doesn't have any graphics and video hardware accelerator, but by default in meta-freescale (meta-freescale/conf/machine/include/imx-base.inc:MACHINE_FIRMWARE_append_use-mainline-bsp) it is being included.

Reply:
I suspect that the following lines are not needed:
```
PACKAGE_EXCLUDE += "firmware-imx-vpu-imx6d"
PACKAGE_EXCLUDE += "firmware-imx-vpu-imx6q"
```
The `MACHINE_EXTRA_RRECOMMENDS_remove` only should do the trick.